src/main/java/de/rub/nds/scanner/core/probe/ScannerProbe.java

```diff
@@ -282,155 +100,49 @@ public final void putIfEqual(
         }
     }
 
-    public final void putIfEqual(
-            AnalyzedProperty determiningProperty,
-            AnalyzedProperty propertyToSet,
-            Map<?, ?> actualResult,
-            String notApplicableReason,
-            TestResult expectedValue) {
-        if (!propertiesMap.containsKey(determiningProperty)
-                || propertiesMap.get(determiningProperty) == null
-                || !propertiesMap.get(determiningProperty).equals(expectedValue)) {
-            put(propertyToSet, new NotApplicableResult(propertyToSet, notApplicableReason));
-        } else {
-            put(propertyToSet, actualResult);
+    private TestResult convertToResult(AnalyzedProperty property, Object result) {
```

Comment:
I have the feeling that converting this method to a public static method of the `TestResult` class is more suitable. This would allow for an easier conversion to test results in other parts of the scanner, for example afterprobes.

Reply:
Yes, you coudl do something like `TestResult.of(Object x)` here

Reply:
But that would mean that TestResult becomes aware of its subclasses (a bit weird)